composer/trainer/mosaic_fsdp.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Released under BSD 3-Clause License,
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+"""Updates FSDPs _auto_wrap to enable module_kwargs and custom process_group cache."""
+
+import functools
+import warnings
+from typing import Any, Callable, Dict, Set, Tuple, Union, cast
+
+import torch
+import torch.nn as nn
+from torch import distributed
+from torch.distributed import ProcessGroup
+from torch.distributed.fsdp import (BackwardPrefetch, CPUOffload, FullyShardedDataParallel, MixedPrecision,
+                                    ShardingStrategy)
+from torch.distributed.fsdp._utils import _contains_batchnorm, _override_batchnorm_mixed_precision
+from torch.distributed.fsdp.wrap import _or_policy, _wrap, _wrap_batchnorm_individually
+
+from composer.core import Precision
+from composer.utils import dist
+
+__all__ = [
+    'sharding_map',
+    'backward_prefetch_map',
+    'get_torch_dtype',
+    '_get_mixed_precision',
```

Comment:
these aren't really private functions since they're being exported, so maybe remove the `_` and add docstrings?